gradio/analytics.py

```diff
@@ -178,29 +178,34 @@ def launched_analytics(blocks: gradio.Blocks, data: dict[str, Any]) -> None:
         [],
     )
 
-    from gradio.blocks import BlockContext
-
     for x in list(blocks.blocks.values()):
-        blocks_telemetry.append(x.get_block_name()) if isinstance(
-            x, BlockContext
-        ) else blocks_telemetry.append(str(x))
-
+        blocks_telemetry.append(x.get_block_name())
     for x in blocks.fns:
         targets_telemetry = targets_telemetry + [
             # Sometimes the target can be the Blocks object itself, so we need to check if its in blocks.blocks
-            str(blocks.blocks[y[0]])
+            blocks.blocks[y[0]].get_block_name()
             for y in x.targets
             if y[0] in blocks.blocks
         ]
         events_telemetry = events_telemetry + [
             y[1] for y in x.targets if y[0] in blocks.blocks
         ]
         inputs_telemetry = inputs_telemetry + [
-            str(blocks.blocks[y]) for y in x.inputs if y in blocks.blocks
+            blocks.blocks[y].get_block_name() for y in x.inputs if y in blocks.blocks
         ]
         outputs_telemetry = outputs_telemetry + [
-            str(blocks.blocks[y]) for y in x.outputs if y in blocks.blocks
+            blocks.blocks[y].get_block_name() for y in x.outputs if y in blocks.blocks
```

Comment:
does `get_block_name` return a string different from before? If so, when comparing analytics this could break things

Reply:
Before we were doing the wrong thing imo. We were storing the full class path `<gradio.components.textbox.Textbox object at 0x11c035790>`.

Change introduced in this PR: https://github.com/gradio-app/gradio/pull/7712 probably